src/main/app/controller/users/EditUserController.ts

```diff
@@ -0,0 +1,135 @@
+import {AuthedRequest} from '../../../types/AuthedRequest';
+import {Response} from 'express';
+import autobind from 'autobind-decorator';
+import {User} from '../../../types/User';
+import {EditUserPageData, SearchUserPageData} from '../../../types/UserPageData';
+import {AxiosError} from 'axios';
+
+@autobind
+export class EditUserController {
+
+  searchErrorMsg = 'A problem occurred when searching for the user. '
+  editErrorMsg = 'A problem occurred when editing the user. '
+  userNotFoundErrorMsg = 'No account was found with the email address:'
+
+  public async renderSearchUser(req: AuthedRequest,
+    res: Response,
+    updated = false,
+    userRolesRemoved = false,
+    userEmail= '',
+    errors: { text: string }[] = []): Promise<void> {
+
+    const pageData: SearchUserPageData = {
+      userEmail: userEmail,
+      errors: errors,
+      updated: updated,
+      userRolesRemoved: userRolesRemoved
+    };
+
+    res.render('users/tabs/searchUserContent', pageData);
+  }
+
+  public async renderEditUser(req: AuthedRequest,
+    res: Response,
+    updated = false,
+    user: User = null,
+    errors: { text: string }[] = []): Promise<void> {
+
+    const pageData: EditUserPageData = {
+      errors: errors,
+      user : user
+    };
+    return res.render('users/tabs/editUserContent', pageData);
+  }
+
+  public async getUser(req: AuthedRequest,
+    res: Response): Promise<void> {
+    const userEmail: string = req.query.userEmail as string;
+
+    await req.scope.cradle.idamApi.getUserByEmail(userEmail, req.session.user.access_token)
+      .then((returnedUser: User) => {
+        if (returnedUser === undefined) {
+          return this.renderSearchUser(req, res, false, false, userEmail, [{ text: `${this.userNotFoundErrorMsg} ${userEmail}` }]);
+        }
+        this.renderEditUser(req,res, false, returnedUser, []);
+      })
+      .catch(async (reason: AxiosError) => {
+        return await this.renderSearchUser(req, res, false, false, userEmail,[{ text: this.searchErrorMsg }]);
+      });
+  }
+
+  public async patchUser(req: AuthedRequest,
+    res: Response,
+    errors: { text: string }[] = [],
+    user: User = null): Promise<void> {
+
+    const userId = req.body.userId;
+    const userEmail = req.body.userEmail;
+    const forename = req.body.forename;
+    const surname = req.body.surname;
+    const role = req.body.role.valueOf();
+    user = await req.scope.cradle.idamApi.getUserByEmail(userEmail, req.session.user.access_token);
+    const roleToRemove = this.getUserRole(user) === '' ? '' : this.getUserRoleToRemove(role[0]['name']);
+
+    await req.scope.cradle.idamApi.updateUserDetails(userId, forename, surname, req.session.user.access_token)
```

Comment:
If a person comes in and only changes the roles, would this call be performed, if so, would it be necessary? 

Would it be better to compare the form coming in with what detail exist for the user and decide what call to make based on that?

Reply:
I've added a check to see if the name has been updated